### PR TITLE
Locations modal

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://ethyca.atlassian.net/browse/PROD-1493

### Description Of Changes

See overall loom in this PR: https://github.com/ethyca/fides/pull/4530

### Code Changes

* [x] Add subgroup modal
* [x] Change location cards to only show locations that do not belong to another location ("parents")
* [x] Make subgroup modal show locations that _do_ belong to another location ("children") UNLESS there are no subgroups (i.e. South America does not currently have subgroups)
* [x] Add some cascading behavior—if you select all of the locations inside "Canada", the "Canada" checkbox will toggle on too. If you deselect one of them, it will toggle off.
* [x] Fix the number selected badge to only count the number of "children" regions
* [x] Make search bring "children" regions up to the card level. This doesn't entirely work yet (if you click "Quebec", "Canada" will also get selected, but if you deselect "Canada", "Quebec" will not get deselected. I didn't implement that yet because I wanted to confirm with design that this is the intended behavior first
* [x] Cypress tests

### Steps to Confirm

* [ ] Run fidesplus off of https://github.com/ethyca/fidesplus/pull/1269
* [ ] Run the admin-ui in dev mode
* [ ] Play around with the location checkboxes and make sure things feel natural!

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/admin-ui/cypress/e2e/locations.cy.ts [hunk 1]
```diff
@@ -28,13 +28,13 @@ describe("Locations", () => {
       cy.getByTestId("picker-card-Europe").within(() => {
         cy.getByTestId("select-all").should("not.be.checked");
         assertIsChecked("European Economic Area", false);
-        assertIsChecked("France", true);
```
Comment (allisonking) [inline clients/admin-ui/cypress/e2e/locations.cy.ts:31]: I updated the fixture to use more realistic data, so these tests had to be updated too

File: clients/admin-ui/src/features/common/PickerCard.tsx [hunk 2]
```diff
@@ -105,12 +128,14 @@ const PickerCard = <T extends { id: string; name: string }>({
             ))}
           </CheckboxGroup>
         </VStack>
-        <Spacer />
-        {showViewMore ? (
```
Comment (allisonking) [inline clients/admin-ui/src/features/common/PickerCard.tsx:109]: the design was updated such that view more button is always present

File: clients/admin-ui/src/features/locations/LocationManagement.tsx [hunk 3]
```diff
@@ -13,8 +13,7 @@ import { usePatchLocationsRegulationsMutation } from "./locations.slice";
 import { groupByContinent } from "./transformations";
 
 const SEARCH_FILTER = (location: Location, search: string) =>
-  location.name?.toLocaleLowerCase().includes(search.toLocaleLowerCase()) ||
-  location.continent?.toLocaleLowerCase().includes(search.toLocaleLowerCase());
+  location.name?.toLocaleLowerCase().includes(search.toLocaleLowerCase());
```
Comment (allisonking) [inline clients/admin-ui/src/features/locations/LocationManagement.tsx:16]: I thought having the continent search was useful, but when I was playing with it I found it confusing. there were no clear reqs on search

File: clients/admin-ui/src/features/locations/LocationPickerCard.tsx [hunk 4]
```diff
@@ -32,29 +42,73 @@ const LocationPickerCard = ({
...
+      if (!newSelections.has(oldSelection)) {
+        locations.forEach((location) => {
+          if (location.belongs_to?.includes(oldSelection)) {
+            newSelections.delete(location.id);
+          }
+        });
+      }
+    });
+
+    // TODO: in "all" view, clicking all of a children will not make the parent selected
```
Comment (allisonking) [inline clients/admin-ui/src/features/locations/LocationPickerCard.tsx:73]: this req is still unclear, waiting for Simon's feedback

**Comment by allisonking** [comment]
Actually I'm unassigning @TheAndrewJackson for now since there are some adjustments that need to be made after meeting with Michael and Simon this morning!

**Review by Kelsey-Ethyca — APPROVED** [review]
🦖

**Pull request merged**